### DISCUSSION TITLE
404 wrong path (subdirectory install)

### DISCUSSION THREAD

**Discussion by Astranor** [givanz/Vvveb#152] [Q&A]
Hello

![1720516848152](https://github.com/givanz/Vvveb/assets/175106685/6e87650a-400e-4d8b-bb48-93d8b8860e8e)

Installed into a subdirectory (vvveb/) and getting 404s, it keeps sending me to root.

I checked answers for similar problems and changed the suggested files in all of them (env.php, .htaccess, index.php, admin/index.php) to no avail.

What other files do I need to edit to make the install work ?

Thanks in advance!

**Comment by givanz** [comment]
Hi

Does it work if you access the install page directly http://www.example.com/vvveb/install/index.php?

For subdirectory install you only need to edit [env.php](https://github.com/givanz/Vvveb/blob/master/env.php#L47) and set `V_SUBDIR_INSTALL` as the subdirectory name, for example if installed in `/vvveb` folder you need to set it to

```php
defined('V_SUBDIR_INSTALL') || define('V_SUBDIR_INSTALL', '/vvveb');
```

If you are using apache you also need to update [.htaccess](https://github.com/givanz/Vvveb/blob/master/.htaccess#L42) line 42 and [/public/.htaccess](https://github.com/givanz/Vvveb/blob/master/public/.htaccess#L43) line 43 and add the `subdirectory/` path before `index.php`

```
RewriteRule ^ vvveb/index.php [L]
```

You don't need to edit index.php or admin/index.php

**Reply by Astranor** [reply]
When accessed directly, I could process with the installation, but I get 404 for most pages after completion, and when I go to the /vvveb url, it systematically sends me to root/install/index.php even though it was already installed.

I tried with a fresh install, only modifying env.php; then also modifying htaccess and public/htaccess, but I am still getting 404 in both cases.

I am on a shared host with Cpanel, tried with PHP 8.1 & PHP 8.3.

Is there maybe a specific function of PHP that needs to be activated ?

**Comment by givanz** [comment]
> When accessed directly, I could process with the installation, but I get 404 for most pages after completion, and when I go to the /vvveb url, it systematically sends me to root/install/index.php even though it was already installed.

Can you please check if `/vvveb/config/db.php` file exists? if install completes successfuly it should create the file, if the file does not exist it will redirect to install page.

One possible cause could be file permissions, probably the file can't be created because there is no write permission for `/config` folder.

---

If you access www.example.com/vvveb/index.php and www.example.com/vvveb/admin/index.php does it work?

if www.example.com/vvveb/index.php works but www.example.com/vvveb/ does not work then probably you don't have apache mod_autoindex enabled.

You need to have mod_autoindex and mod_rewrite apache modules enabled.

**Reply by Astranor** [reply]
Replied as a new answer by mistake, it's just below.

**Comment by Astranor** [comment]
So, upon testing more with a new install, first off just changing env.php still sends to root when accessing www.website.com/vvveb, but, when accessing www.website.com/vvveb/install/index.php, the install process works, and creates config/db.php as required.

The second problem occurs once the installation process is complete, it sends me to "www.website.com/vvveb/vvveb/admin//index.php?success=Installation%20succesful!&errors=" which gives a 404, because there's "vvveb" twice instead of once.

The third issue is, then accessing www.website.com/vvveb stills sends to root, which is why it tries to go to www.website.com/install/index.php, because from root directory it cannot find config/db.php is my guess.

If I manually edit the url and go to "www.website.com/vvveb/admin//index.php?success=Installation%20succesful!&errors=", I then get a installation successful and can connect to the admin panel, but the Home page still points to root directory, and thus other pages below all throw 404.

![1720539464054](https://github.com/givanz/Vvveb/assets/175106685/e7cdcdd8-c3ee-4a38-a0cd-535f85b58171)

edit: Forgot to mention, from that point if I try to access www.website.com/vvveb/index.php it lets me see the page, though it throws 404 for all content.

**Comment by givanz** [comment]
Thanks for feedback.

Double subdirectory in redirect was a bug, it's fixed in the latest.zip changed file `/install/controller/index.php`

If it works with www.website.com/vvveb/index.php but not with www.website.com/vvveb/ then this is probably caused by some apache configuration related to mod_autoindex or mod_rewrite

Make sure that you have mod_autoindex and mod_rewrite active by accesing a php file with

```php
<?php
phpinfo();
```

They should show in the apache section.

![mod_rewrite](https://github.com/givanz/Vvveb/assets/302738/8e2786f8-4cec-4b25-983e-5dafc4e12889)

The pages that show not found message have correct urls eg http://www.website.com/vvveb/page/about?

If they have wrong url's can you try clearing cache from admin?

Just to make sure, you need to have the subdirectory path set in env.php withtout a trailing slash eg `/vvveb/` it should be `/vvveb`

**Comment by Astranor** [comment]
Hello, thanks for assisting.

I checked everything you mentioned to no avail.

mod_autoindex and mod_rewrite are both active.
env.php's subdirectory path does not have a trailing slash or other issues.
Pages' urls match your description, but there is no "page" folder in the vvveb folder, I suppose because during install it tried to create these using the wrong path? Should there be a check for folders during install, and throw an error if folders are not found?

**Comment by givanz** [comment]
Hi

page is not a folder, page urls are captured through mod rewrite that sends all page requests to index.php, the rewrite rules are inside .htaccess

Another thing to check is that apache has `AllowOverride All` enabled and `.htaccess` is processed.

You can check this by writing something random inside .htaccess to generate an error, after the change you should get a 500 internal server error if `.htaccess` is loaded if not then your configuration does not load .htaccess file.

Some other things to try is to change `/vvveb` subdirectory file permissions and add execute flag.

You can also check apache error log to see what is generating 404 error when accessing page urls.
